platformifier/templates/generic/.environment

```diff
@@ -32,9 +32,20 @@ export DATABASE_PORT="$DB_PORT"
 export DATABASE_NAME="$DB_PATH"
 export DATABASE_USERNAME="$DB_USERNAME"
 export DATABASE_PASSWORD="$DB_PASSWORD"
-export DATABASE_SCHEME="$DB_SCHEME"
+export DATABASE_SCHEME="$DB_CONNECTION"
 
-export ADMIN_JWT_SECRET="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
-export JWT_SECRET="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
-export API_TOKEN_SALT="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
+# Set secrets needed by Strapi, if they are not set
+# Prefer setting these as project secret variables with {{ .Assets.Binary }} variable:create env:SECRET_NAME --sensitive=true
+if [[ -z "$ADMIN_JWT_SECRET" ]]; then
+  export ADMIN_JWT_SECRET="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
+fi
+if [[ -z "$JWT_SECRET" ]]; then
+  export JWT_SECRET="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
+fi
+if [[ -z "$API_TOKEN_SALT" ]]; then
+    export API_TOKEN_SALT="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
+fi
+if [[ -z "$APP_KEYS" ]]; then
+    export APP_KEYS="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
```

Comment:
The `APP_KEYS` needs to have 2 random strings separated by a comma and here you just add 1 :( 
this should be: 
```
export APP_KEYS="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY,${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
```

Reply:
as i'm not sure how to define 2 different random strings and as @gilzow is right saying that random string could be have the same value, you can duplicate the PROJECT_ENTROPY ;)

Reply:
One key is required, multiple keys are used for rotating keys (the first key is always used for signing and any more keys are only used for verification, so that you can safely rotate the key and at some point decide to invalidate old keys).

You can read more about this in the following links:

* Koa `app.keys` (underlying framework): https://koajs.com/
* Keygrip (underlying library): https://github.com/crypto-utils/keygrip

> The keylist is an array of all valid keys for signing, in descending order of freshness; new keys should be unshifted into the array and old keys should be popped.

> keys.sign(data)
> This creates a SHA1 HMAC based on the first key in the keylist, and outputs it as a 27-byte url-safe base64 digest (base64 without padding, replacing + with - and / with _).